### PR TITLE
messy first pass

### PR DESCRIPTION
**Pull request by ttheune**
First pass at updates for allowing proxy failovers.

### CODE REVIEW

File: linux/centos/7/foxpass_setup.py [hunk 1]
```diff
@@ -38,16 +38,22 @@ def main():
     parser.add_argument('--bind-pw', required=True, help='Bind Password')
     parser.add_argument('--api-key', required=True, help='API Key')
     parser.add_argument('--ldap-uri', default='ldaps://ldap.foxpass.com', help='LDAP Server')
```
Comment (aren) [inline linux/centos/7/foxpass_setup.py]: Add alias for --primary-ldap ?

File: linux/centos/7/foxpass_setup.py [hunk 2]
```diff
@@ -38,16 +38,22 @@ def main():
...
+    parser.add_argument('--secondary-ldap', dest='ldaps', default=[], action='append', help='Secondary LDAP Server(s)')
     parser.add_argument('--api-url', default='https://api.foxpass.com', help='API Url')
```
Comment (aren) [inline linux/centos/7/foxpass_setup.py]: Add alias for --primary-api?

File: linux/centos/7/foxpass_setup.py [hunk 3]
```diff
@@ -38,16 +38,22 @@ def main():
...
     bind_dn = 'cn=%s,%s' % (args.bind_user, args.base_dn)
 
+    uris = [args.ldap_uri]
+    for uri in args.ldaps:
+        uris.append(uri)
+
     install_dependencies()
     write_foxpass_ssh_keys_script(args.api_url, args.api_key)
-    run_authconfig(args.ldap_uri, args.base_dn)
+    run_authconfig(uri, args.base_dn)
```
Comment (aren) [inline linux/centos/7/foxpass_setup.py]: uris?

File: linux/centos/7/foxpass_setup.py [hunk 4]
```diff
@@ -99,8 +105,8 @@ def write_foxpass_ssh_keys_script(api_url, api_key):
         os.system('chmod 700 /usr/local/bin/foxpass_ssh_keys.sh')
 
 
-def run_authconfig(uri, base_dn):
-    cmd = 'authconfig --enablesssd --enablesssdauth --enablelocauthorize --enableldap --enableldapauth --ldapserver={uri} --disableldaptls --ldapbasedn={base_dn} --enablemkhomedir --enablecachecreds --update'.format(uri=uri, base_dn=base_dn)
+def run_authconfig(uris, base_dn):
+    cmd = 'authconfig --enablesssd --enablesssdauth --enablelocauthorize --enableldap --enableldapauth --ldapserver={uri} --disableldaptls --ldapbasedn={base_dn} --enablemkhomedir --enablecachecreds --update'.format(uri=join(uris), base_dn=base_dn)
```
Comment (aren) [inline linux/centos/7/foxpass_setup.py]: ','.join(uris)

File: linux/debian/8/foxpass_setup.py [hunk 5]
```diff
@@ -39,19 +39,31 @@ def main():
...
+    parser.add_argument('--secondary-ldap', dest='ldaps', default=[], action='append', help='Secondary LDAP Server(s)')
+    parser.add_argument('--api-url', '--api', default='https://api.foxpass.com', help='API Url')
+    parser.add_argument('--secondary-api', dest='apis', default=[], action='append', help='Secondary API Server(s)')
     parser.add_argument('--ldap-connections', default=2, help='Number of connections to make to LDAP server.')
 
     args = parser.parse_args()
 
     binddn = 'cn=%s,%s' % (args.bind_user, args.base_dn)
 
+    uris = ['uri %s' % args.ldap_uri]
```
Comment (aren) [inline linux/debian/8/foxpass_setup.py]: style: send in the array of uris to write_nslcd_conf, and let it handle the formatting (adding the uri prefix).

File: linux/debian/8/foxpass_setup.py [hunk 6]
```diff
@@ -39,19 +39,31 @@ def main():
...
+    for uri in args.ldaps:
+        uris.append('uri %s' % uri)
+
+    curls = ['curl -s -q -m 5 -f "%s/sshkeys/?secret=${secret}&user=${user}&hostname=${hostname}&aws_instance_id=${aws_instance_id}" 2>/dev/null' % args.api_url]
```
Comment (aren) [inline linux/debian/8/foxpass_setup.py]: style: send in the list of api urls to write_foxpass_ssh_keys_script, and let it handle the curl stuff.

also, don't repeat this long curl string. store it in a variable once, and use that.
curl_pattern = 'curl -s -q -m 5 -f "%s/sshkeys/?secret=${secret}&user=${user}&hostname=${hostname}&aws_instance_id=${aws_instance_id}" 2>/dev/null'
...
apis.append(curl_pattern % api)

File: linux/debian/8/foxpass_setup.py [hunk 7]
```diff
@@ -39,19 +39,31 @@ def main():
...
+
+    seperator = ' || '
+
     apt_get_update()
     install_dependencies()
-    write_foxpass_ssh_keys_script(args.api_url, args.api_key)
-    write_nslcd_conf(uri=args.ldap_uri, basedn=args.base_dn, binddn=binddn, bindpw=args.bind_pw,
-                     threads=int(args.ldap_connections))
+    write_foxpass_ssh_keys_script(curls, args.api_key, seperator)
+    write_nslcd_conf(uris=uris, secondary_ldap=secondary_ldap, basedn=args.base_dn,
```
Comment (aren) [inline linux/debian/8/foxpass_setup.py]: secondary_ldap is not used. move separator variable into write_foxpass_ssh_keys_script function (and fix spelling).

File: linux/debian/8/foxpass_setup.py (hunk 6 above)
Comment (aren) [inline linux/debian/8/foxpass_setup.py]: also note that there are two different curl_patterns used

File: linux/centos/7/foxpass_setup.py (hunk 1 above)
Comment (ttheune) [inline linux/centos/7/foxpass_setup.py]: Not in this version of the script yet (note no changes to *many* other copies).

File: linux/centos/7/foxpass_setup.py (hunk 2 above)
Comment (ttheune) [inline linux/centos/7/foxpass_setup.py]: Not in this version of the script yet (note no changes to *many* other copies).

File: linux/centos/7/foxpass_setup.py (hunk 3 above)
Comment (ttheune) [inline linux/centos/7/foxpass_setup.py]: yup, fixed typo

File: linux/centos/7/foxpass_setup.py (hunk 4 above)
Comment (ttheune) [inline linux/centos/7/foxpass_setup.py]: thanks, copy pasta fail.

File: linux/debian/8/foxpass_setup.py (hunk 5 above)
Comment (ttheune) [inline linux/debian/8/foxpass_setup.py]: fixed

File: linux/debian/8/foxpass_setup.py (hunk 6 above)
Comment (ttheune) [inline linux/debian/8/foxpass_setup.py]: doh, fixed

File: linux/debian/8/foxpass_setup.py (hunk 7 above)
Comment (ttheune) [inline linux/debian/8/foxpass_setup.py]: fixed

File: linux/debian/8/foxpass_setup.py [hunk 8]
```diff
@@ -49,21 +49,20 @@ def main():
...
     binddn = 'cn=%s,%s' % (args.bind_user, args.base_dn)
 
-    uris = ['uri %s' % args.ldap_uri]
+    uris = [args.ldap_uri]
     for uri in args.ldaps:
-        uris.append('uri %s' % uri)
+        uris.append(uri)
 
-    curls = ['curl -s -q -m 5 -f -H "Authorization: Token ${secret}" "%s/sshkeys/?user=${user}&hostname=${hostname}&aws_instance_id=${aws_instance_id}" 2>/dev/null' % args.api_url]
+    base_curl = 'curl -s -q -m 5 -f -H "Authorization: Token ${secret}" "%s/sshkeys/?user=${user}&hostname=${hostname}'
```
Comment (aren) [inline linux/debian/8/foxpass_setup.py]: Move this section inside of write_foxpass_ssh_keys_script

File: linux/centos/7/foxpass_setup.py [hunk 9]
```diff
@@ -51,9 +51,14 @@ def main():
     for uri in args.ldaps:
         uris.append(uri)
 
+    base_curl = 'curl -s -q -m 5 -f -H "Authorization: Token ${secret}" "%s/sshkeys/?user=${user}&hostname=${hostname}'
+    curls = [base_curl % args.api_url]
+    for api in args.apis:
+        curls.append(base_curl % api)
+
     install_dependencies()
     write_foxpass_ssh_keys_script(args.api_url, args.api_key)
```
Comment (aren) [inline linux/centos/7/foxpass_setup.py]: this takes the wrong (old) arguments still.

File: linux/debian/8/foxpass_setup.py (hunk 8 above)
Comment (ttheune) [inline linux/debian/8/foxpass_setup.py]: moved

File: linux/centos/7/foxpass_setup.py (hunk 9 above)
Comment (ttheune) [inline linux/centos/7/foxpass_setup.py]: re-built based on deb8 example
also moved "uris" inside of configure_sssd

File: linux/debian/8/foxpass_setup.py [hunk 10]
```diff
@@ -49,9 +51,9 @@ def main():
 
     apt_get_update()
     install_dependencies()
-    write_foxpass_ssh_keys_script(args.api_url, args.api_key)
```
Comment (aren) [inline linux/debian/8/foxpass_setup.py]: combine args.api_url and args.apis into one array to send into write_foxpass_ssh_keys_script()

File: linux/debian/8/foxpass_setup.py [hunk 11]
```diff
@@ -49,9 +51,9 @@ def main():
...
-    write_nslcd_conf(uri=args.ldap_uri, basedn=args.base_dn, binddn=binddn, bindpw=args.bind_pw,
-                     threads=int(args.ldap_connections))
+    write_foxpass_ssh_keys_script(args.api_url, args.apis, args.api_key)
+    write_nslcd_conf(uri=args.ldap_uri, ldaps=args.ldaps, basedn=args.base_dn,
```
Comment (aren) [inline linux/debian/8/foxpass_setup.py]: combine args.ldap_uri and args.ldaps into one array to send into write_nslcd_conf()

File: linux/centos/7/foxpass_setup.py [hunk 12]
```diff
@@ -77,9 +77,9 @@ def write_foxpass_ssh_keys_script(api_url, apis, api_key):
     for api in apis:
         curls.append(base_curl % api)
 
-    with open('/usr/local/bin/foxpass_ssh_keys.sh', "w") as w:
+    with open('/usr/sbin/foxpass_ssh_keys.sh', "w") as w:
```
Comment (aren) [inline linux/centos/7/foxpass_setup.py]: if you want to move this script to sbin, you need to update the reference to it in augment_sshd_config

File: linux/debian/8/foxpass_setup.py (hunk 10 above)
Comment (ttheune) [inline linux/debian/8/foxpass_setup.py]: This is done... not sure why it's still showing up here :(

File: linux/debian/8/foxpass_setup.py [hunk 13]
```diff
@@ -39,18 +39,22 @@ def main():
...
     binddn = 'cn=%s,%s' % (args.bind_user, args.base_dn)
+    apis = [args.api_url] + args.apis
+    uris = [args.ldap_uri] + args.ldaps
 
     apt_get_update()
     install_dependencies()
-    write_foxpass_ssh_keys_script(args.api_url, args.api_key)
-    write_nslcd_conf(uri=args.ldap_uri, basedn=args.base_dn, binddn=binddn, bindpw=args.bind_pw,
+    write_foxpass_ssh_keys_script(apis, args.api_key)
+    write_nslcd_conf(uri=uris, basedn=args.base_dn, binddn=binddn, bindpw=args.bind_pw,
```
Comment (aren) [inline linux/debian/8/foxpass_setup.py]: write_nslcd_conf(uris, ...

**Pull request closed without merging**